test/dsl/custom_functions.jl

```diff
@@ -33,32 +35,13 @@ let
         v5 * (X7^2) / (K5^2 + X7^2 + Y7^2), X7 + Y7 --> Z7
     end
 
-    function permute_ps(pvals, rn1, rn2)
-        ps1 = parameters(rn1)
-        ps2 = parameters(rn2)
-        pvals2 = similar(pvals)
-        for (i, p) in enumerate(ps2)
-            pidx = findfirst(isequal(p), ps1)
-            pvals2[i] = pvals[pidx]
-        end
-        pvals2
-    end
-
-    f1 = ODEFunction(convert(ODESystem, custom_function_network_1), jac = true)
-    f2 = ODEFunction(convert(ODESystem, custom_function_network_2), jac = true)
-    g1 = SDEFunction(convert(SDESystem, custom_function_network_1))
-    g2 = SDEFunction(convert(SDESystem, custom_function_network_2))
     for factor in [1e-2, 1e-1, 1e0, 1e1, 1e2]
-        u0 = factor * rand(rng, length(get_unknowns(custom_function_network_1)))
-        p = factor * rand(rng, length(get_ps(custom_function_network_2)))
-
-        # needed as this code assumes an ordering of the parameters and species...
-        p2 = permute_ps(p, custom_function_network_1, custom_function_network_2)
-
+        u0 = rnd_u0(custom_function_network_1, rng; factor)
```

Comment:
Using the new functionality and helper functions, quite a few tests can be written more concisely, which is nice.